src/utils.rs

```diff
@@ -707,25 +707,26 @@ impl<'a, 'b> fmt::Display for Emoji<'a, 'b> {
 }
 
 fn str_width(s: &str) -> usize {
-    #[cfg(feature = "unicode-width")]
+    #[cfg(feature = "unicode-display-width")]
     {
-        use unicode_width::UnicodeWidthStr;
-        s.width()
+        unicode_display_width::width(s) as usize
     }
-    #[cfg(not(feature = "unicode-width"))]
+    #[cfg(not(feature = "unicode-display-width"))]
     {
         s.chars().count()
     }
 }
 
 #[cfg(feature = "ansi-parsing")]
 fn char_width(c: char) -> usize {
-    #[cfg(feature = "unicode-width")]
+    #[cfg(feature = "unicode-display-width")]
     {
-        use unicode_width::UnicodeWidthChar;
-        c.width().unwrap_or(0)
+        match unicode_display_width::is_double_width(c) {
```

Comment:
For compatibility, maybe we should replicate the logic from `unicode-width` here for control characters? Maybe like:

```rust
match c.is_control() {
     true if c == '\x00' => ?,
     true => 0,
     false => match unicode_display_width::is_double_width(c) {
         true => 2,
         false => 1,
     }
}
```